indigo/analysis/refs/provisions.py

```diff
@@ -0,0 +1,555 @@
+        while ref.element is None and roots:
+            ref.element = self.find_numbered_hier_element(roots[-1], names, ref.text)
+            if ref.element is None:
+                # no match, try again with a higher root
+                roots.pop(-1)
+
+        if ref.element is not None:
+            ref.eId = ref.element.get('eId')
+            if ref.child:
+                self.resolve_ref(ref.child, [ref.element])
+
+    def find_numbered_hier_element(self, root: Element, names: Union[List[str], None], num: str, above_root: bool = False) -> Element:
+        """Find an heir element with the given number. Looks for elements with the given names, or any hier element
+        if names is None. If above_root is True, then we'll look above the root element if nothing inside matches.
+        """
+        names = names or AkomaNtoso30.hier_elements
+        ns = root.nsmap[None]
+
+        # prefix with namespace
+        names = [f'{{{ns}}}{n}' for n in names]
+        dead_ends = [f'{{{ns}}}{n}' for n in ['quotedStructure', 'embeddedStructure', 'content']]
+
+        # do a breadth-first search, starting at root, and walk upwards, expanding until we find something or reach the top
+        for elem in bfs_upward_search(root, names, dead_ends, above_root):
+            # ignore matches to the root element, which avoids things like section (1)(a)(a) matching the same (a) twice
+            if elem == root:
+                continue
+            num_elem = elem.find('a:num', {'a': ns})
+            if num_elem is not None and num_elem.text.rstrip(".") == num:
+                return elem
+
+
+class ProvisionRefsMatcher(TextPatternMatcher):
+    """ Finds internal references to sections in documents, of the form:
+
+        # singletons
+        section 26
+        section 26B
+
+        # lists
+        sections 22 and 32
+        and sections 19, 22 and 23, unless it appears to him
+        and sections 19, 22, and 23 (oxford comma)
+        and sections 19,22 and 23 (incorrect spacing)
+        Sections 24, 26, 28, 36, 42(2), 46, 48, 49(2), 52, 53, 54 and 56 shall mutatis mutandis
+        sections 23, 24, 25, 26 and 28;
+        sections 22(1) and 25(3)(b);
+        sections 18, 61 and 62(1).
+        in terms of section 2 or 7
+        A person who contravenes sections 4(1) and (2), 6(3), 10(1) and (2), 11(1), 12(1), 19(1), 19(3), 20(1), 20(2), 21(1), 22(1), 24(1), 25(3), (4) , (5) and (6) , 26(1), (2), (3) and (5), 28(1), (2) and (3) is guilty of an offence.
+    """
+    xml_marker_tag = "ref"
+    xml_ancestor_xpath = '|'.join(f'//ns:{x}'
+                                  for x in ['coverpage', 'preface', 'preamble', 'body', 'mainBody', 'judgmentBody', 'conclusions'])
+    xml_candidate_xpath = (".//text()[not("
+                           "ancestor::ns:ref or ancestor::ns:heading or ancestor::ns:subheading or ancestor::ns:num "
+                           "or ancestor::ns:embeddedStructure or ancestor::ns:quotedStructure)]")
```

Comment:
```suggestion
                           "or ancestor::ns:embeddedStructure or ancestor::ns:quotedStructure or ancestor::ns:table)]")
```

Reply:
a bit rough n ready, but will avoid us marking up section references in tables where the amended Act is given in a different cell (normally in the first column)

Reply:
added